pkg/jobs/jobs.go

```diff
@@ -0,0 +1,128 @@
+package jobs
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const (
+	// Queued state
+	Queued State = "queued"
+	// Running state
+	Running = "running"
+	// Errored state
+	Errored = "errored"
+)
+
+const (
+	// JSONEncoding is a JSON encoding message type
+	JSONEncoding = "json"
+	// GobEncoding is a Gob encoding message type
+	GobEncoding = "gob"
+)
+
+type (
+	// Queue interface is used to represent a asynchronous queue of jobs from
+	// which it is possible to enqueue and consume jobs.
+	Queue interface {
+		Enqueue(*Job) error
+		Consume() (bool, *Job, error)
+		Foreach(func(*Job) error) error
```

Comment:
I'll remove it. It is indeed of no use for now.

In the jobs doc I documented the possibility for a user to fetch all the element of the queue but I'm not sure we really want to allow that.